src/main/app/controller/courts/CourtTypesController.ts

```diff
@@ -0,0 +1,142 @@
+
+    }
+
+    let allCourtTypes: CourtType[] = [];
+
+    await req.scope.cradle.api.getCourtTypes()
+      .then((value: CourtType[]) => allCourtTypes = value)
+      .catch(() => error += this.getCourtTypesErrorMsg);
+
+
+    const pageData: CourtTypePageData = {
+      errorMsg: error,
+      updated: updated,
+      items: this.mapCourtTypeToCourtTypeItem(allCourtTypes, courtCourtTypes)
+
+    };
+
+    res.render('courts/tabs/typesContent', pageData);
+  }
+
+
+  public async put(req: AuthedRequest, res: Response): Promise<void> {
+
+    let courtCourtTypes: CourtType[] = null;
+
+    if (req.body.types) {
+
+      courtCourtTypes = this.mapBodyToCourtType(req.body);
+
+      if(courtCourtTypes.find( c => (c.name ==="Magistrates' Court" && !c.code ) || (c.name ==="Magistrates' Court" && isNaN(c.code))
+      || (c.name==='County Court' && !c.code) || (c.name ==='County Court' && isNaN(c.code))
+      || (c.name === 'Crown Court' && !c.code)  || (c.name ==='Crown Court' && isNaN(c.code)))){
+
+        return this.get(req, res, false, this.emptyCourtCodeErrorMsg, courtCourtTypes);
+      }
+
+      else
+      {
+        await req.scope.cradle.api.updateCourtCourtTypes(req.params.slug, courtCourtTypes)
+          .then((value: CourtType[]) => this.get(req, res, true, '', value))
+          .catch(() => this.get(req, res, false, this.updateErrorMsg, courtCourtTypes));
+      }
+    }
+    else
+    {
+      return this.get(req, res, false, this.emptyCourtTypesErrorMsg, courtCourtTypes);
+    }
+
+  }
+
+  private mapCourtTypeToCourtTypeItem(allCourtTypes: CourtType[], courtCourtTypes: CourtType[]): CourtTypeItem[] {
+
+    if( courtCourtTypes && allCourtTypes) {
+      const courtTypeItems = allCourtTypes.map((ott: CourtType) => (
+        {
+          value: ott.id + ',' + ott.name,
+          text: ott.name,
+          checked: this.isChecked(ott, courtCourtTypes),
+          code: this.getCode(ott.id, courtCourtTypes)
+        }));
+
+      return courtTypeItems;
+    }
+    else
+      return [];
+  }
+
+
+  private mapBodyToCourtType(body: any): CourtType[] {
+
+    const courtTypes: string[] = Array.isArray(body.types) ? body.types : [body.types];
+
+    const courtTypeItems = courtTypes.map((ct) => (
+      {
+        id: parseInt(ct.split(',',2)[0]),
+        name: ct.split(',',2)[1],
+        code:this.setCode(ct.split(',',2)[1], body.magistratesCourtCode, body.countyCourtCode, body.crownCourtCode),
+
+      }));
+
+    return courtTypeItems;
+  }
+
+
+
+  private isChecked(courtType: CourtType, courtCourtTypes: CourtType[]) {
+    return (courtCourtTypes.some(e => e.id === courtType.id));
+  }
+
+  private getCode(id: number, courtCourtTypes: CourtType[]) {
+
+    return (courtCourtTypes.find(e => e.id === id) ? courtCourtTypes.find(e => e.id === id).code : null);
+
+  }
+
+  private setCode(name: string, magistratesCourtCode: string, countyCourtCode: string, crownCourtCode: string){
+
+    const regExp = /[a-zA-Z]/g;
```

Comment:
Is this just ensuring the code is a number? Do we need to catch punctuation etc?

Reply:
Replaced with checking if string has only number(regExp = /^[0-9]*$/;) and alter logic accordingly